### DISCUSSION TITLE
Hlines and vlines with color result in transparent line and corner

### DISCUSSION THREAD

**Discussion by libinxu01** [lvjr/tabularray#375] [Q&A]
I set some cells, vlines and hlines with background color, and find transparent lines and corners. How to fix it ?
![企业微信截图_58544cad-d5fb-4b42-a49d-13f55b869ef7](https://user-images.githubusercontent.com/15907992/222134927-d42f749b-47fe-42f2-861c-1b2be3f02a02.png)
![企业微信截图_204c909a-0e17-4b8f-bc41-7aad33d2811f](https://user-images.githubusercontent.com/15907992/222134948-e05b4187-165d-437d-9f5a-1baaa6c10eca.png)

```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xespotcolor}
\usepackage{xcolor}

\definecolor{bgColorFourColorRed}   {cmyk}{0,    0.11,  0.07,   0}


\begin{document}
	\begin{center}
		\begin{longtblr}{
				width=\dimexpr\textwidth\relax + 5mm, 
				colspec={X[l,m]X[l,m]X[l,m]X[l,m]}, 
				cells={font={\fontsize{0.025\textwidth}{0.025\textwidth}\selectfont}, bg=bgColorFourColorRed}, 
				hlines={1.0pt}, 
				vlines={1.0pt},
				hline{1, Z}={1.0pt,fg=bgColorFourColorRed},
				hline{2-Y}={1,Z}{1.0pt,fg=bgColorFourColorRed},
				vline{1,Z}={1.0pt,fg=bgColorFourColorRed},
				vline{2-Y}={1,Z}{1.0pt,fg=bgColorFourColorRed},
				rowsep=0pt, 
				row{1}={5.6690pt}, row{2}={0.1149\textwidth}, row{3}={0.1149\textwidth}, row{4}={0.1149\textwidth}, row{5}={0.1149\textwidth}, row{6}={5.6690pt}, 
				rowfoot=1, 
			}
			\SetCell[r=1,c=4]{halign=c,valign=h}{}&&&\\* 
			\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}\\ 
			\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}\\ 
			\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}\\ 
			\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}&\SetCell[r=1,c=1]{halign=c,valign=h}{}\\ 
			\SetCell[r=1,c=4]{halign=c,valign=h}{}&&&\\* 
		\end{longtblr}
	\end{center}
\end{document}
```

**Comment by muzimuzhi** [comment]
> and find transparent lines and corners

Not reproducible in both texstudio's built-in poppler-based PDF viewer and the free Adobe Reader on macOS. Thus I infer it's related to specific PDF readers.

This fixes the not-crossing issue:
```diff
- hline{2-Y}={1,Z}{1.0pt,fg=bgColorFourColorRed},
+ hline{2-Y}={1,Z}{1.0pt,fg=bgColorFourColorRed, leftpos=0, rightpos=0},
```

```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage{xcolor}
\definecolor{bgColorFourColorRed}   {cmyk}{0,    0.11,  0.07,   0}

\begin{document}
\begin{tblr}{
  hlines={3.0pt}, 
  vlines={3.0pt},
  hline{1,Z}=     {fg=bgColorFourColorRed},
  hline{2-Y}={1,Z}{fg=bgColorFourColorRed,leftpos=0, rightpos=0},
  vline{1,Z}=     {fg=bgColorFourColorRed},
  vline{2-Y}={1,Z}{fg=bgColorFourColorRed},
}
  a & b & c & d \\
  a & b & c & d \\
  a & b & c & d \\
\end{tblr}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/222225819-3be36275-63d8-43be-93c7-5a2b60271d58.png)


Please put more effort in simplifying your example.

**Reply by libinxu01** [reply]
> > and find transparent lines and corners
> 
> Not reproducible in both texstudio's built-in poppler-based PDF viewer and the free Adobe Reader on macOS. Thus I infer it's related to specific PDF readers.
> 
> This fixes the not-crossing issue:
> 
> ```diff
> - hline{2-Y}={1,Z}{1.0pt,fg=bgColorFourColorRed},
> + hline{2-Y}={1,Z}{1.0pt,fg=bgColorFourColorRed, leftpos=0, rightpos=0},
> ```
> 
> ```
> \documentclass{article}
> \usepackage{tabularray}
> \usepackage{xcolor}
> \definecolor{bgColorFourColorRed}   {cmyk}{0,    0.11,  0.07,   0}
> 
> \begin{document}
> \begin{tblr}{
>   hlines={3.0pt}, 
>   vlines={3.0pt},
>   hline{1,Z}=     {fg=bgColorFourColorRed},
>   hline{2-Y}={1,Z}{fg=bgColorFourColorRed,leftpos=0, rightpos=0},
>   vline{1,Z}=     {fg=bgColorFourColorRed},
>   vline{2-Y}={1,Z}{fg=bgColorFourColorRed},
> }
>   a & b & c & d \\
>   a & b & c & d \\
>   a & b & c & d \\
> \end{tblr}
> \end{document}
> ```
> 
> ![image](https://user-images.githubusercontent.com/6376638/222225819-3be36275-63d8-43be-93c7-5a2b60271d58.png)
> 
> Please put more effort in simplifying your example.

Thanks so much!
